[CORE-514] Always return compact entity provider

samanehsan opened the pull request:
Ticket: https://broadworkbench.atlassian.net/browse/CORE-514

- Update EntityManager to always return `CompactEntityProvider`
- Remove unused parameters from `defaultEntityManager`
- Remove `isCompactDataTablesEnabled` from EntityService (was called for workspace cloning, and when enabled was applied to newly created workspaces)

To-Do:
- Remove optional `workspaceSettingRepository` parameter from EntityService (probably should be a separate PR)
- Update workspace settings to remove compact entity setting (there is a separate ticket for this now)
---

**PR checklist**

- [x] Include the JIRA issue number in the PR description and title
- [ ] Make sure Swagger is updated if API changes
  - [ ] **...and Orchestration's Swagger too!**
- [ ] If you changed anything in `model/`, then you should publish a new official `rawls-model` and perform the corresponding dependency updates as specified in the [README](https://github.com/broadinstitute/rawls/blob/develop/README.md#publish-rawls-model):
  - [ ] in the automation subdirectory
  - [ ] in workbench-libs
  - [ ] in firecloud-orchestration
- [ ] Get two thumbsworth of PR review
- [ ] Verify all tests go green, including CI tests
- [ ] **Squash commits and merge** to develop (branches are automatically deleted after merging)
- [ ] Inform other teams of any substantial changes via Slack and/or email

samanehsan committed:
Always return compact entity provider

samanehsan committed:
Merge branch 'develop' of github.com:broadinstitute/rawls into se/CORE-514-remove-entity-manager-ws-settings-check

samanehsan committed:
Remove unused defaultEntityManager args

samanehsan committed:
Remove checking ws compact entity settings

samanehsan committed:
Update tests

samanehsan committed:
Merge branch 'develop' of github.com:broadinstitute/rawls into se/CORE-514-remove-entity-manager-ws-settings-check

samanehsan committed:
Remove code comment

davidangb reviewed (approved):
lgtm, modulo one unit test to fix. :+1: assuming you fix that test!

samanehsan committed:
Fix failing test

We no longer need to separately test
that a workspace was created with compact
data tables because now that is the default.

samanehsan commented:
The diff looks confusing, but I just deleted `"cloneWorkspace" should "create a V2 Workspace using compact data tables"` because it would otherwise look the same as `"cloneWorkspace" should "create a Workspace"` now that every workspace uses compact data tables.
